src/middleware/product/productExists.js

```diff
@@ -0,0 +1,21 @@
+import { collectionServices, productsServices } from '../../services';
+
+const isProductSeller = async (req, res, next) => {
+  const product = await productsServices.getProductById(req.params.id);
+  if (!product) {
+    return res.status(404).json({
+      code: '404',
+      message: 'Failed',
+      error: 'Product not found !!!',
```

Comment:
I like the return statement inside IF. However, I think it would be better if you threw an error instead of returning a JSON response. As in creating an NotFoundError and UnauthourisedError in the helpers then import them like

`
import { NotFoundError, UnauthorizedError } from '../../helpers/errors';
--- 
if (!product) {
      throw new NotFoundError('Product not found');
    }
    ---
    if (req.user.id !== collection.userId) {
      throw new UnauthorizedError('User is not authorized to sell this product');
    }
    `

Reply:
Throwing an error insider an asynchronous middleware requires a try and catch for it to be handled by express error handler. that means we would need to create another wrap to all async middleware. that idea is good to have an error file which has all errors.

Reply:
404 not found is not an error response